parsers/CR.py

```diff
@@ -179,85 +66,97 @@ def fetch_production(
 ):
     # ensure we have an arrow object.
     # if no target_datetime is specified, this defaults to now.
-    target_datetime = arrow.get(target_datetime).to(TIMEZONE)
+    arw_datetime = arrow.get(target_datetime).to(TIMEZONE)
 
     # if before 01:30am on the current day then fetch previous day due to
     # data lag.
     today = arrow.get().to(TIMEZONE).date()
-    if target_datetime.date() == today:
-        target_datetime = (
-            target_datetime
-            if target_datetime.time() >= time(1, 30)
-            else target_datetime.shift(days=-1)
+    if arw_datetime.date() == today:
+        arw_datetime = (
+            arw_datetime
+            if arw_datetime.time() >= time(1, 30)
+            else arw_datetime.shift(days=-1)
         )
 
-    if target_datetime < arrow.get("2012-07-01"):
+    if arw_datetime < arrow.get("2012-07-01"):
         # data availability limit found by manual trial and error
         logger.error(
             "CR API does not provide data before 2012-07-01, "
-            "{} was requested".format(target_datetime),
+            "{} was requested".format(arw_datetime),
             extra={"key": zone_key},
         )
         return None
 
     # Do not use existing session as some amount of cache is taking place
     r = Session()
-    url = "https://apps.grupoice.com/CenceWeb/CencePosdespachoNacional.jsf"
+    day = arw_datetime.format("DD")
+    month = arw_datetime.format("MM")
+    year = arw_datetime.format("YYYY")
+    url = f"https://apps.grupoice.com/CenceWeb/data/sen/json/EnergiaHorariaFuentePlanta?anno={year}&mes={month}&dia={day}"
+
     response = r.get(url)
+    resp_data = response.json()["data"]
 
-    soup = BeautifulSoup(response.text, "html.parser")
-    jsf_view_state = soup.find("input", {"name": "javax.faces.ViewState"})["value"]
+    results = {}
+    for hourly_item in resp_data:
+        if hourly_item["fuente"] == "Intercambio":
+            continue
+        if hourly_item["fecha"] not in results:
+            results[hourly_item["fecha"]] = empty_record(zone_key)
 
-    data = [
-        ("formPosdespacho:txtFechaInicio_input", target_datetime.format(DATE_FORMAT)),
-        ("formPosdespacho:pickFecha", ""),
-        ("formPosdespacho_SUBMIT", 1),
-        ("javax.faces.ViewState", jsf_view_state),
-    ]
-    response = r.post(url, data=data)
+        results[hourly_item["fecha"]]["datetime"] = arrow.get(
+            hourly_item["fecha"], tzinfo=TIMEZONE
+        ).datetime
 
-    # tell pandas which table to use by providing CHARACTERISTIC_NAME
-    df = pd.read_html(
-        response.text, match=CHARACTERISTIC_NAME, skiprows=1, index_col=0
-    )[0]
+        if (
+            SPANISH_TO_ENGLISH[hourly_item["fuente"]]
+            not in results[hourly_item["fecha"]]["production"]
+        ):
+            results[hourly_item["fecha"]]["production"][
+                SPANISH_TO_ENGLISH[hourly_item["fuente"]]
+            ] = 0.0
```

Comment:
Why are we manually assigning this to 0?
It should be either ignored or raise a exception if it's not available.

Right now it seems to result in hydro being 0 even when the data is unavailabe, we should avoid this so we can run estimations.